relayer/processor/metrics.go

```diff
@@ -62,5 +68,9 @@ func NewPrometheusMetrics() *PrometheusMetrics {
 			Name: "cosmos_relayer_fees_spent",
 			Help: "The amount of fees spent from the relayer's wallet",
 		}, walletLabels),
+		TxFailureError: registerer.NewCounterVec(prometheus.CounterOpts{
+			Name: "cosmos_relayer_tx_failure",
```

Comment:
Similar to a previous comment, `cosmos_relayer_tx_errors_total` might be a better name. The `err` label better be replaced with `cause`  or `type`. Fewer words are better used as the value instead of the full error message to avoid "Cardinality bombs".

https://promlabs.com/blog/2022/12/11/avoid-these-6-mistakes-when-getting-started-with-prometheus/#mistake-1-cardinality-bombs

Reply:
`cosmos_relayer_tx_failures_total` is also a good name.

Prometheus uses this convention for its own errors.
<img width="600" alt="image" src="https://github.com/cosmos/relayer/assets/1257310/3a365ed0-144f-4e77-bcc5-25d9571f26cc">

Reply:
Cool, agree. Used `caused` in this case.